### PR TITLE
Refactor availability controller

### PR DESCRIPTION
**Pull request by hackmastera**
This is a refactor pulling business logic out of the availability controller, moving it with its tests to a `VoyagerLookup` class.

This is work toward making space for the alma side of this functionality, as well as being a step toward #266.

Available for review now or I can work on top of this to get 266 implemented; it just seemed like it might be nice to get it in sooner rather than later.

### CODE REVIEW

**Review by carolyncole — APPROVED** [review]
@hackmastera It was hard to determine if your changes had broken anything since the tests were also changed so drastically.  I went ahead and reverted your changes to the controller test and was happy to say they all still passed.  Looks good to me.  I will not merge since @christinach wants to review.

File: spec/models/voyager_lookup_spec.rb [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        bib_id = '9468468'
+        allow(VoyagerHelpers::Liberator).to receive(:get_availability).and_return("9329199"=>{ more_items: false, location: "f", status: "Order Received 12-16-2015" })
+        availability = described_class.single_bib_availability(bib_id: bib_id)
+        _holding, details = availability.first
+        expect(details[:status]).to include('Order Received')
+      end
+    end
+
+    context 'when given a record with an elf (electronic) location' do
+      it 'returns a status of online' do
```
Comment (christinach) [inline spec/models/voyager_lookup_spec.rb:110]: This should say `elf1` not `elf`

**Comment by hackmastera** [comment]
> @hackmastera It was hard to determine if your changes had broken anything since the tests were also changed so drastically. I went ahead and reverted your changes to the controller test and was happy to say they all still passed. Looks good to me. I will not merge since @christinach wants to review.

Awesome; thank you for double-checking.

**Review by christinach — CHANGES_REQUESTED** [review]
It looks great. I added some comments.

File: spec/models/voyager_lookup_spec.rb [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+      bib_id = '929437'
+      availability = described_class.single_bib_availability(bib_id: bib_id)
+      expect(availability).to eq availability_hash
+    end
+  end
+
+  describe '#multiple_bib_availability' do
+    it 'provides partial availability' do
+      bib_id = '929437'
+      allow(VoyagerHelpers::Liberator).to receive(:get_availability).and_return(bib_id => {})
```
[the remark this answers is not in the document]
Comment (christinach) [reply spec/models/voyager_lookup_spec.rb:32]: This could also be added in a different PR.

**Comment by hackmastera** [comment]
@christinach thanks for the review; I believe I've addressed the comments. ready for another look.

**Pull request merged**